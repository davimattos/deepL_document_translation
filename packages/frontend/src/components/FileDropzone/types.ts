export type Props = {
  handleDrop: (e: React.DragEvent) => void
  handleDragOver: (e: React.DragEvent) => void
  file: File | null
  fileInputRef: HTMLInputElement | any
  handleFileSelect: (selectedFile: File) => void
  allowedExtensions: string[]
}