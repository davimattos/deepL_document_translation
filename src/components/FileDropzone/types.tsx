export type Props = {
  handleDrop: (e: React.DragEvent) => void
  handleDragOver: (e: React.DragEvent) => void
  file: File | null
  fileInputRef: HTMLInputElement | any
  handleFileSelect: (selectedFile: File) => void
  handleExecutedFile: () => void
  allowedExtensions: string[]
}