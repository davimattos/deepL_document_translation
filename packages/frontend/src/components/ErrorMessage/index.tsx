import { AlertCircle } from "lucide-react";
import { Props } from "./types";

export function ErrorMessage({ errorMessage }: Props) {
  return (
    <div className="bg-red-50 border border-red-200 rounded-lg p-4">
      <div className="flex items-start gap-3">
        <AlertCircle
          data-testid="alert-icon"
          className="w-5 h-5 text-red-500 mt-0.5 flex-shrink-0"
        />
        <p className="text-red-700 text-sm">{errorMessage}</p>
      </div>
    </div>
  );
}
