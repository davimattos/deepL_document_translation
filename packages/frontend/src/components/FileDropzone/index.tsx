import { Upload, CheckCircle } from "lucide-react";
import { Props } from "./types";

export function FileDropzone({
  handleDrop,
  handleDragOver,
  file,
  fileInputRef,
  handleFileSelect,
  allowedExtensions,
}: Props) {
  return (
    <div className="p-8">
      <div
        onDrop={handleDrop}
        onDragOver={handleDragOver}
        className={`border-2 border-dashed rounded-xl p-12 text-center transition-all duration-300 ${
          file
            ? "border-green-300 bg-green-50"
            : "border-gray-300 bg-gray-50 hover:border-blue-400 hover:bg-blue-50"
        }`}
      >
        <input
          ref={fileInputRef}
          type="file"
          accept={allowedExtensions.toString()}
          onChange={e => e.target.files?.[0] && handleFileSelect(e.target.files[0])}
          className="hidden"
        />

        {file ? (
          <div className="space-y-4">
            <CheckCircle className="w-16 h-16 text-green-500 mx-auto" />
            <div>
              <h3 className="text-xl font-semibold text-gray-900 mb-2">Arquivo Selecionado</h3>
              <p className="text-gray-600 mb-1">{file.name}</p>
              <p className="text-sm text-gray-500">{(file.size / 1024 / 1024).toFixed(2)} MB</p>
            </div>
            <button
              onClick={() => fileInputRef.current?.click()}
              className="text-blue-600 hover:text-blue-700 font-medium transition-colors"
            >
              Selecionar outro arquivo
            </button>
          </div>
        ) : (
          <div className="space-y-4">
            <Upload className="w-16 h-16 text-gray-400 mx-auto" />
            <div>
              <h3 className="text-xl font-semibold text-gray-900 mb-2">Carregar Documento</h3>
              <p className="text-gray-600 mb-4">
                Arraste e solte seu arquivo aqui ou clique para selecionar
              </p>
              <button
                onClick={() => fileInputRef.current?.click()}
                className="bg-blue-600 hover:bg-blue-700 text-white px-6 py-3 rounded-lg font-medium transition-colors"
              >
                Selecionar Arquivo
              </button>
            </div>
            <div className="text-sm text-gray-500">
              <p className="mb-2">Extens√µes suportadas:</p>
              <p className="font-mono text-xs">{allowedExtensions.join(", ")}</p>
            </div>
          </div>
        )}
      </div>
    </div>
  );
}
