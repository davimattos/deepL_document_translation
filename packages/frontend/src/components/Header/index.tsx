import { FileText } from "lucide-react";

export function Header() {
  return (
    <div className="text-center mb-12">
      <div className="flex items-center justify-center mb-6">
        <div className="bg-blue-600 p-3 rounded-xl">
          <FileText role="img" aria-label="Icone de documento" className="w-8 h-8 text-white" />
        </div>
      </div>
      <h1 className="text-4xl font-bold text-gray-900 mb-4">
        Tradutor de Documentos
      </h1>
      <p className="text-xl text-gray-600 max-w-2xl mx-auto">
        Traduza seus documentos usando a API do DeepL
      </p>
    </div>
  );
}
