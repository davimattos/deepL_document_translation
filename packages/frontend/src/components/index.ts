export * from './Header'
export * from './LanguageSelector'
export * from './FileDropzone'
export * from './ProcessingProgress'
export * from './InfoCard'
export * from './ErrorMessage'