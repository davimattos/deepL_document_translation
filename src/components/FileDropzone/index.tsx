import { Upload, CheckCircle } from "lucide-react";
import { Props } from "./types";

export function FileDropzone({
  handleDrop,
  handleDragOver,
  file,
  fileInputRef,
  handleFileSelect,
  allowedExtensions,
  handleExecutedFile
}: Props) {

  return (
    <div className="p-8">
      <div
        onDrop={handleDrop}
        onDragOver={handleDragOver}
        className={`border-2 border-dashed rounded-xl p-12 text-center transition-all duration-300 ${
          file
            ? "border-green-300 bg-green-50"
            : "border-gray-300 bg-gray-50 hover:border-blue-400 hover:bg-blue-50"
        }`}
      >
        <input
          ref={fileInputRef}
          type="file"
          accept=".docx,.pptx,.xlsx,.pdf,.htm,.html,.txt,.xlf,.xliff,.srt"
          onChange={(e) =>
            e.target.files?.[0] && handleFileSelect(e.target.files[0])
          }
          className="hidden"
        />

        {file ? (
          <div className="space-y-4">
            <CheckCircle className="w-16 h-16 text-green-500 mx-auto" />
            <div>
              <h3 className="text-xl font-semibold text-gray-900 mb-2">
                Arquivo Selecionado
              </h3>
              <p className="text-gray-600 mb-1">{file.name}</p>
              <p className="text-sm text-gray-500">
                {(file.size / 1024 / 1024).toFixed(2)} MB
              </p>
            </div>
            <button
              onClick={handleExecutedFile}
              className="text-blue-600 hover:text-blue-700 font-medium transition-colors"
            >
              Selecionar outro arquivo
            </button>
          </div>
        ) : (
          <div className="space-y-4">
            <Upload className="w-16 h-16 text-gray-400 mx-auto" />
            <div>
              <h3 className="text-xl font-semibold text-gray-900 mb-2">
                Carregar Documento
              </h3>
              <p className="text-gray-600 mb-4">
                Arraste e solte seu arquivo aqui ou clique para selecionar
              </p>
              <button
                onClick={() => fileInputRef.current?.click()}
                className="bg-blue-600 hover:bg-blue-700 text-white px-6 py-3 rounded-lg font-medium transition-colors"
              >
                Selecionar Arquivo
              </button>
            </div>
            <div className="text-sm text-gray-500">
              <p className="mb-2">Extens√µes suportadas:</p>
              <p className="font-mono text-xs">
                {allowedExtensions.join(", ")}
              </p>
            </div>

            {/* File Format Limits Table */}
            <div className="mt-6 bg-gray-50 rounded-lg p-4">
              <h4 className="font-semibold text-gray-900 mb-3">
                Limites por Formato de Arquivo
              </h4>
              <div className="overflow-x-auto">
                <table className="w-full text-xs">
                  <thead>
                    <tr className="border-b border-gray-200">
                      <th className="text-center py-2 font-medium text-gray-700">
                        Formato
                      </th>
                      <th className="text-center py-2 font-medium text-gray-700">
                        DeepL Suporte
                      </th>
                    </tr>
                  </thead>
                  <tbody className="text-gray-600">
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">Word (.docx / .doc) </td>
                      <td className="text-center py-2">
                        30 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">PowerPoint (.pptx) </td>
                      <td className="text-center py-2">
                        30 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">Excel (.xlsx) </td>
                      <td className="text-center py-2">
                        30 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">PDF (.pdf) </td>
                      <td className="text-center py-2">
                        30 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">Text (.txt)</td>
                      <td className="text-center py-2">
                        1 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">HTML (.html)</td>
                      <td className="text-center py-2">
                        5 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">XLIFF (.xlf/.xliff)*</td>
                      <td className="text-center py-2">
                        10 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr>
                      <td className="py-2 font-mono">SRT (.srt) </td>
                      <td className="text-center py-2">
                        150 KB
                        <br />
                        1M chars
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <p className="text-xs text-gray-500 mt-3">
                * Os limites aplicam-se por documento individual
              </p>
            </div>
          </div>
        )}
      </div>
    </div>
  );
}
