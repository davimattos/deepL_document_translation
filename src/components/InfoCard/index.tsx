import { Props } from "./types";

export function InfoCard({icon, title, description, className = 'bg-gray-100'}: Props) {
  return (
    <div className="bg-white p-6 rounded-xl shadow-sm border border-gray-100">
      <div className={`w-12 h-12 rounded-lg flex items-center justify-center mb-4 ${className}`}>
        {icon}
      </div>
      <h3 className="font-semibold text-gray-900 mb-2">{title}</h3>
      <p className="text-gray-600 text-sm">
        {description}
      </p>
    </div>
  );
}
