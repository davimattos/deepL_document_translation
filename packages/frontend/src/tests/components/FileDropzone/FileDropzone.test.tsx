import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, vi, expect } from 'vitest';

import { FileDropzone } from '../../../components/FileDropzone';
import React from 'react';

describe('FileDropzone', () => {
  const mockHandleDrop = vi.fn();
  const mockHandleDragOver = vi.fn();
  const mockHandleFileSelect = vi.fn();
  const mockHandleExecutedFile = vi.fn();
  const mockRef = { current: { click: vi.fn() } };

  const allowedExtensions = ['.docx', '.pdf', '.txt'];

  it('should render the empty dropzone with instructions and button', () => {
    render(
      <FileDropzone
        handleDrop={mockHandleDrop}
        handleDragOver={mockHandleDragOver}
        file={null}
        fileInputRef={mockRef as any}
        handleFileSelect={mockHandleFileSelect}
        allowedExtensions={allowedExtensions}
        handleExecutedFile={mockHandleExecutedFile}
      />
    );

    expect(screen.getByRole('heading', { name: /carregar documento/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /selecionar arquivo/i })).toBeInTheDocument();
    expect(screen.getByText(/extensões suportadas/i)).toBeInTheDocument();

    // snapshot da versão vazia
    expect(document.body).toMatchSnapshot();
  });

  it('should open file input when "Selecionar Arquivo" button is clicked', () => {
    const inputRef: any = React.createRef<HTMLInputElement>();

    render(
      <FileDropzone
        handleDrop={mockHandleDrop}
        handleDragOver={mockHandleDragOver}
        file={null}
        fileInputRef={inputRef}
        handleFileSelect={mockHandleFileSelect}
        allowedExtensions={allowedExtensions}
        handleExecutedFile={mockHandleExecutedFile}
      />
    );

    expect(inputRef.current).not.toBeNull();

    const clickSpy = vi.spyOn(inputRef.current!, 'click');

    const button = screen.getByRole('button', { name: /selecionar arquivo/i });
    fireEvent.click(button);
    expect(clickSpy).toHaveBeenCalled();
    clickSpy.mockRestore();
  });

  it('should render selected file information', () => {
    const fakeFile = new File(['dummy content'], 'documento.pdf', { type: 'application/pdf' });

    render(
      <FileDropzone
        handleDrop={mockHandleDrop}
        handleDragOver={mockHandleDragOver}
        file={fakeFile}
        fileInputRef={mockRef as any}
        handleFileSelect={mockHandleFileSelect}
        allowedExtensions={allowedExtensions}
        handleExecutedFile={mockHandleExecutedFile}
      />
    );

    expect(screen.getByRole('heading', { name: /arquivo selecionado/i })).toBeInTheDocument();
    expect(screen.getByText('documento.pdf')).toBeInTheDocument();
    expect(screen.getByText(/selecionar outro arquivo/i)).toBeInTheDocument();

    // snapshot da versão com arquivo
    expect(document.body).toMatchSnapshot();
  });

  it('should call handleExecutedFile when "Selecionar outro arquivo" is clicked', () => {
    const fakeFile = new File(['dummy content'], 'exemplo.txt', { type: 'text/plain' });

    render(
      <FileDropzone
        handleDrop={mockHandleDrop}
        handleDragOver={mockHandleDragOver}
        file={fakeFile}
        fileInputRef={mockRef as any}
        handleFileSelect={mockHandleFileSelect}
        allowedExtensions={allowedExtensions}
        handleExecutedFile={mockHandleExecutedFile}
      />
    );

    const button = screen.getByRole('button', { name: /selecionar outro arquivo/i });
    fireEvent.click(button);
    expect(mockHandleExecutedFile).toHaveBeenCalled();
  });
});
