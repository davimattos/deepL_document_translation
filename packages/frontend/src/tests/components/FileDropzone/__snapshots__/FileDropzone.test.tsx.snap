// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FileDropzone > should render selected file information 1`] = `
<body>
  <div>
    <div
      class="p-8"
    >
      <div
        class="border-2 border-dashed rounded-xl p-12 text-center transition-all duration-300 border-green-300 bg-green-50"
      >
        <input
          accept=".docx,.pptx,.xlsx,.pdf,.htm,.html,.txt,.xlf,.xliff,.srt"
          class="hidden"
          type="file"
        />
        <div
          class="space-y-4"
        >
          <svg
            class="lucide lucide-check-circle w-16 h-16 text-green-500 mx-auto"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
            />
            <path
              d="m9 11 3 3L22 4"
            />
          </svg>
          <div>
            <h3
              class="text-xl font-semibold text-gray-900 mb-2"
            >
              Arquivo Selecionado
            </h3>
            <p
              class="text-gray-600 mb-1"
            >
              documento.pdf
            </p>
            <p
              class="text-sm text-gray-500"
            >
              0.00
               MB
            </p>
          </div>
          <button
            class="text-blue-600 hover:text-blue-700 font-medium transition-colors"
          >
            Selecionar outro arquivo
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`FileDropzone > should render the empty dropzone with instructions and button 1`] = `
<body>
  <div>
    <div
      class="p-8"
    >
      <div
        class="border-2 border-dashed rounded-xl p-12 text-center transition-all duration-300 border-gray-300 bg-gray-50 hover:border-blue-400 hover:bg-blue-50"
      >
        <input
          accept=".docx,.pptx,.xlsx,.pdf,.htm,.html,.txt,.xlf,.xliff,.srt"
          class="hidden"
          type="file"
        />
        <div
          class="space-y-4"
        >
          <svg
            class="lucide lucide-upload w-16 h-16 text-gray-400 mx-auto"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
            />
            <polyline
              points="17 8 12 3 7 8"
            />
            <line
              x1="12"
              x2="12"
              y1="3"
              y2="15"
            />
          </svg>
          <div>
            <h3
              class="text-xl font-semibold text-gray-900 mb-2"
            >
              Carregar Documento
            </h3>
            <p
              class="text-gray-600 mb-4"
            >
              Arraste e solte seu arquivo aqui ou clique para selecionar
            </p>
            <button
              class="bg-blue-600 hover:bg-blue-700 text-white px-6 py-3 rounded-lg font-medium transition-colors"
            >
              Selecionar Arquivo
            </button>
          </div>
          <div
            class="text-sm text-gray-500"
          >
            <p
              class="mb-2"
            >
              Extens√µes suportadas:
            </p>
            <p
              class="font-mono text-xs"
            >
              .docx, .pdf, .txt
            </p>
          </div>
          <div
            class="mt-6 bg-gray-50 rounded-lg p-4"
          >
            <h4
              class="font-semibold text-gray-900 mb-3"
            >
              Limites por Formato de Arquivo
            </h4>
            <div
              class="overflow-x-auto"
            >
              <table
                class="w-full text-xs"
              >
                <thead>
                  <tr
                    class="border-b border-gray-200"
                  >
                    <th
                      class="text-center py-2 font-medium text-gray-700"
                    >
                      Formato
                    </th>
                    <th
                      class="text-center py-2 font-medium text-gray-700"
                    >
                      DeepL Suporte
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="text-gray-600"
                >
                  <tr
                    class="border-b border-gray-100"
                  >
                    <td
                      class="py-2 font-mono"
                    >
                      Word (.docx / .doc) 
                    </td>
                    <td
                      class="text-center py-2"
                    >
                      30 MB
                      <br />
                      1M chars
                    </td>
                  </tr>
                  <tr
                    class="border-b border-gray-100"
                  >
                    <td
                      class="py-2 font-mono"
                    >
                      PowerPoint (.pptx) 
                    </td>
                    <td
                      class="text-center py-2"
                    >
                      30 MB
                      <br />
                      1M chars
                    </td>
                  </tr>
                  <tr
                    class="border-b border-gray-100"
                  >
                    <td
                      class="py-2 font-mono"
                    >
                      Excel (.xlsx) 
                    </td>
                    <td
                      class="text-center py-2"
                    >
                      30 MB
                      <br />
                      1M chars
                    </td>
                  </tr>
                  <tr
                    class="border-b border-gray-100"
                  >
                    <td
                      class="py-2 font-mono"
                    >
                      PDF (.pdf) 
                    </td>
                    <td
                      class="text-center py-2"
                    >
                      30 MB
                      <br />
                      1M chars
                    </td>
                  </tr>
                  <tr
                    class="border-b border-gray-100"
                  >
                    <td
                      class="py-2 font-mono"
                    >
                      Text (.txt)
                    </td>
                    <td
                      class="text-center py-2"
                    >
                      1 MB
                      <br />
                      1M chars
                    </td>
                  </tr>
                  <tr
                    class="border-b border-gray-100"
                  >
                    <td
                      class="py-2 font-mono"
                    >
                      HTML (.html)
                    </td>
                    <td
                      class="text-center py-2"
                    >
                      5 MB
                      <br />
                      1M chars
                    </td>
                  </tr>
                  <tr
                    class="border-b border-gray-100"
                  >
                    <td
                      class="py-2 font-mono"
                    >
                      XLIFF (.xlf/.xliff)*
                    </td>
                    <td
                      class="text-center py-2"
                    >
                      10 MB
                      <br />
                      1M chars
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="py-2 font-mono"
                    >
                      SRT (.srt) 
                    </td>
                    <td
                      class="text-center py-2"
                    >
                      150 KB
                      <br />
                      1M chars
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <p
              class="text-xs text-gray-500 mt-3"
            >
              * Os limites aplicam-se por documento individual
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
